search/terms_aggregation.go

```diff
@@ -0,0 +1,201 @@
+package search
+
```

Comment:
can we change file name `bucket_terms_aggregation.go` ?

Reply:
I'm not sure that prefix `bucket` is that useful. It's a broad categorization for a family of opensearch aggregations. What value do you think it adds for the file name?

@dylan-bourque @jeffmclean-cs , any thoughts?

Reply:
I'd leave this as is.  The types in this file are focused on terms, not buckets.

Reply:
The terms are defining the buckets 🙃 but we could skip the categorization of `bucket` and just add an `aggs_` prefix to the file maybe.

Reply:
Maybe just `terms.go`?

Reply:
`terms.go` would conflict with the `TermsQuery`, which could be argued to be in the `terms.go` file.

Reply:
That's fair, but I don't think we should start down the path of having a file for each type either.  TBH, I still think the current file name is good.

Reply:
Discussed with @jeffmclean-cs and @dylan-bourque in a call this morning. We'll start the naming scheme `search_aggs_*` with this becoming `search_aggs_terms.go` and moving it to the top level domain package.